### ISSUE TITLE
Pull latest Enki code to enable to OpenID Connect support

### ISSUE THREAD

**Issue by RyanTG** [scottwainstock/pbm-blog#33]
Our old method of logging in to the blog doesn't work. When you try, you get redirected to this: https://support.google.com/accounts/answer/6206245?p=openid&rd=1

But Enki added support to OpenID Connect in this commit: https://github.com/xaviershay/enki/commit/ec85aef26ee82f17e8225be5cbe03ea459d1f112

And the readme has information on setting it up: https://github.com/xaviershay/enki/

Can we update the blog to use this?

(the other option is to use Publify. And they recently fixed the two big bugs that were making publify unusable.)

**Comment by RyanTG** [comment]
I'll attempt to do this myself. If I hit a snag, I'll tell you.

**Comment by RyanTG** [comment]
Since my pbm-blog is forked from you, and you forked it (I'm assuming) from Enki - if I wanted to get the latest Enki commits, would I temporarily change my upstream from you to xaviershay/enki? That seems like a weird thing to do.

Also, I'm stuck trying to fix a stupid bundle error right now. I'm using the right ruby version, so not sure why I'm getting the error...

**Comment by scottwainstock** [comment]
Hmmm... yeah this is a weird one. It seems like maybe you could add a new upstream for xaviershay/enki and the "git rebase UPSTREAM master" on it?

https://help.github.com/articles/configuring-a-remote-for-a-fork/

**Comment by RyanTG** [comment]
I usually just set upstream and remote by editing the config file in .git. It seems to do the trick.

I'm concerned that the latest version of Enki will be using a different version of Rails and Ruby. But I guess the worst thing that happens is I break my local version of the blog...


I fixed my bundle issue. When I cloned my fork of pbm-blog (I didn't have it on this computer), it told me to use ruby-1.9.3p545. BUT bundle install didn't work until I actually used ruby-1.9.3. Strange.

**Comment by RyanTG** [comment]
Whenever you get a chance, can you email me pbm-blog/config/database.yml ? I think I lost mine...

**Comment by RyanTG** [comment]
Shoot, looks like I lost all my code updates in the rebase.

Not sure if it's because I used "git pull --rebase upstream master" instead of what you suggested. I was getting an upstream error. Confusing...

**Comment by RyanTG** [comment]
Duuuuuuuuude. It works. I can log in. Can't believe that actually worked.

Next I'll try to add a second auth for you. I had to generate credentials from https://console.developers.google.com

**Comment by scottwainstock** [comment]
Nice! The rebase with a new upstream worked?

This is kind of a weird way for this guy to package his stuff... but whatever.

**Comment by RyanTG** [comment]
I couldn't get git rebase upstream master to work. Said it was an invalid upstream. Then I tried git pull --rebase upstream (I think), and it pulled everything but overwrote my files. So I reset and then did a plain old git pull upstream master, and there were six conflicted files. So I manually merged those files, and all was good!

One thing I have a question about. For the auth and key, I should be setting those with ENV through heroku. Is the blog within our glowing-dusk app? I figure it is, but wasn't sure.
